src/main/java/adris/altoclef/ui/MessageSender.java

```diff
@@ -1,66 +1,67 @@
 package adris.altoclef.ui;
 
+
 import adris.altoclef.Debug;
 import adris.altoclef.util.csharpisbetter.Timer;
 import net.minecraft.client.MinecraftClient;
 
 import java.util.Comparator;
+import java.util.Objects;
 import java.util.PriorityQueue;
 
+
 /**
- * We can't send messages immediately as the server will kick us.
- * As such, we will send messages in a delayed queued fashion.
+ * We can't send messages immediately as the server will kick us. As such, we will send messages in a delayed queued fashion.
  */
 public class MessageSender {
-
+    // TODO: 2021-05-22 god, this class is horrible (no offense)
```

Comment:
hah none taken. By that you mean the triple nested "if" check for all three timers in the `tick()` method and the re-setting of the timers/counters right? I really should've split those into functions (like `isReadyToSendChat()` or something like that, named better) and maybe had the timers/trackers a subclass or something to prevent dumb duplication.